cpp/open3d/core/nns/KnnIndex.cpp

```diff
@@ -127,16 +135,19 @@ std::pair<Tensor, Tensor> KnnIndex::SearchKnn(const Tensor& query_points,
     dataset_points_, points_row_splits_, query_points_, queries_row_splits_, \
             knn, neighbors_index, neighbors_row_splits, neighbors_distance
 
-#define CALL(type, fn)                                              \
-    if (Dtype::FromType<type>() == dtype) {                         \
-        fn<type, int32_t>(FN_PARAMETERS);                           \
+#define CALL(ftype, itype, fn)                                      \
```

Comment:
We have this error in CI, when CUDA is not enabled:
```
/home/runner/work/Open3D/Open3D/cpp/open3d/core/nns/KnnIndex.cpp:109:17: error: variable ‘index_dtype’ set but not used [-Werror=unused-but-set-variable]
     const Dtype index_dtype = GetIndexDtype();
```

Macros are hard to read. I would do the naive approach here: remove te `#define`  and just code it with `if-else`. It might be slightly longer, but it will be much easier to read.

Reply:
Done.